src/common/telemetry.ts

```diff
@@ -1,377 +1,381 @@
+            if (!optIn) {
+                Telemetry.send(new TelemetryEvent(Telemetry.appName + "/telemetryOptOut"), true);
             }
 
-            /*
-             * Load settings data from settingsHome/TelemetrySettings.json
-             */
-            private static loadSettings(): ITelemetrySettings {
-                try {
-                    TelemetryUtils.telemetrySettings = JSON.parse(<any>fs.readFileSync(TelemetryUtils.telemetrySettingsFile));
-                } catch (e) {
-                    // if file does not exist or fails to parse then assume no settings are saved and start over
-                    TelemetryUtils.telemetrySettings = {};
-                }
+            TelemetryUtils.optInCollectedForCurrentSession = true;
+            TelemetryUtils.saveSettings();
+        }
+
+        private static getUserType(): string {
+            let userType: string = TelemetryUtils.telemetrySettings.userType;
 
-                return TelemetryUtils.telemetrySettings;
+            if (userType === undefined) {
+            if (process.env[TelemetryUtils.INTERNAL_USER_ENV_VAR]) {
+                userType = TelemetryUtils.USERTYPE_INTERNAL;
+            } else {
+                let domain: string = process.env.USERDNSDOMAIN;
+                domain = domain ? domain.toLowerCase().substring(domain.length - TelemetryUtils.INTERNAL_DOMAIN_SUFFIX.length) : null;
+                userType = domain === TelemetryUtils.INTERNAL_DOMAIN_SUFFIX ? TelemetryUtils.USERTYPE_INTERNAL : TelemetryUtils.USERTYPE_EXTERNAL;
             }
 
-            /*
-             * Save settings data in settingsHome/TelemetrySettings.json
-             */
-            private static saveSettings(): void {
-                let settingsHome = HostPlatform.getSettingsHome();
-                if (!fs.existsSync(settingsHome)) {
-                    fs.mkdirSync(settingsHome);
-                }
+            TelemetryUtils.telemetrySettings.userType = userType;
+        }
 
-                fs.writeFileSync(TelemetryUtils.telemetrySettingsFile, JSON.stringify(TelemetryUtils.telemetrySettings));
-            }
+            return userType;
+        }
 
-            private static getUniqueId(regValue: string, regHive: string, fallback: () => string): Q.Promise<any> {
-                let uniqueId: string;
-                if (os.platform() === "win32") {
-                    return TelemetryUtils.getRegistryValue(TelemetryUtils.REGISTRY_SQMCLIENT_NODE, regValue, regHive)
-                    .then(function(id: string): Q.Promise<string> {
-                        if (id) {
-                            uniqueId = id.replace(/[{}]/g, "");
-                            return Q.resolve(uniqueId);
-                        } else {
-                            return Q.resolve(fallback());
-                        }
-                    });
+        private static getRegistryValue(key: string, value: string, hive: string): Q.Promise<string> {

```

Comment:
Do we still need this now that we are using the vscode-extension-telemetry package?